archiver/index.d.ts

```diff
@@ -37,6 +37,7 @@ declare namespace archiver {
         directory(dirpath: string, destpath: nameInterface | string): void;
         directory(dirpath: string, destpath: nameInterface | string, data: any | Function): void;
 
+        file(source: string, name: nameInterface): void;
```

Comment:
`name`  should be called `data`, and should have a type named `EntryData` - so you'll need to export an interface that reflects the following type: https://archiverjs.com/docs/global.html#EntryData